src/EnergyPlus/HVACControllers.cc

```diff
@@ -867,19 +822,14 @@ void InitController(EnergyPlusData &state, int const ControlNum, bool &IsConverg
 
     static constexpr std::string_view RoutineName("InitController");
 
-    int ActuatedNode;
-    int SensedNode;
-    int ControllerIndex;
-    Real64 rho; // local fluid density
-
-    auto &NumControllers(state.dataHVACControllers->NumControllers);
-    auto &InitControllerOneTimeFlag(state.dataHVACControllers->InitControllerOneTimeFlag);
-    auto &InitControllerSetPointCheckFlag(state.dataHVACControllers->InitControllerSetPointCheckFlag);
+    int &NumControllers(state.dataHVACControllers->NumControllers);
+    bool &InitControllerOneTimeFlag(state.dataHVACControllers->InitControllerOneTimeFlag);
+    bool &InitControllerSetPointCheckFlag(state.dataHVACControllers->InitControllerSetPointCheckFlag);
     auto &ControllerProps(state.dataHVACControllers->ControllerProps);
     auto &RootFinders(state.dataHVACControllers->RootFinders);
-    auto &MyEnvrnFlag(state.dataHVACControllers->MyEnvrnFlag);
-    auto &MySizeFlag(state.dataHVACControllers->MySizeFlag);
-    auto &MyPlantIndexsFlag(state.dataHVACControllers->MyPlantIndexsFlag);
+    Array1D_bool &MyEnvrnFlag(state.dataHVACControllers->MyEnvrnFlag);
+    Array1D_bool &MySizeFlag(state.dataHVACControllers->MySizeFlag);
+    Array1D_bool &MyPlantIndexsFlag(state.dataHVACControllers->MyPlantIndexsFlag);
```

Comment:
This is a fine change.  I'll repeat that in the future, it would be great to not have these separate arrays of flags.  Just add a `bool envrnFlag` and `bool sizeFlag` and `bool plantIndexFlag` to the controller structure.  That would eliminate 3 additional arrays that have to live out on state, and likely at least one or two boolean flags that already live on state that are only used to one-time allocate these arrays.  :pushpin: